lib/semantics/check-io.cc

```diff
@@ -706,4 +723,11 @@ void IoChecker::CheckForProhibitedSpecifier(
   }
 }
 
+void IoChecker::CheckForPureSubprogram() const {  // C1597
+  if (context_.location() &&
+      FindPureProcedureContaining(&context_.FindScope(*context_.location()))) {
```

Comment:
Can `context_.location()` ever be null? It looks like this is only called for a *Stmt which is wrapped in a Statement, which would have provided the location.

Reply:
It's an `optional`.  Would you prefer a `CHECK()`?

Reply:
Done.